File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
    int newNum=x;
    int res=0;

    while(x > 0){
        int lD=x%10;
        res=res*10 + lD;
        x=x/10;
    }

       return (newNum==res ? true : false);
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) return false;

        int revHalf = 0;
        while (x > revHalf) {
            int digit = x % 10;
            revHalf = revHalf * 10 + digit;
            x /= 10;
        }

        
        return (x == revHalf || x == revHalf / 10);

    }
};